apps/src/gamelab/animationListModule.js

```diff
@@ -712,7 +712,15 @@ export function animationSourceUrl(key, props, withVersion = false) {
   // 2. Otherwise it's local to this project, and we should use the animation
   //    key to look it up in the animations API.
   return animationsApi.basePath(key) + '.png' +
-      ((withVersion && props.version) ? '?version=' + props.version : '');
+      (
+        (withVersion && props.version)
+        ? '?version=' + props.version
+        // Temporary: Local-cachebust animation requests, so after restoring
+        // and old version of the project the client isn't seeing newer animations.
+        // Should be able to tear this out once we always request animations
+        // by specific version.
+        : '?cachebust=' + Math.floor(Math.random() * 0xFFFFFF).toString(16)
```

Comment:
This is a mitigation for a particular problem I ran into during manual testing:  I could prove that our server was doing all the right things on restore, all files on S3 end up in a good state, etc. but the client would still show the wrong animation version.

This is because the client is still loading animations without a version id - and even though animation `A.png` has a new (restored) version on the server, when the client sees a request for `A.png` it grabs whatever it's got locally cached.  This problem will go away when our client can always ask for animations by version, but in the meantime we'll do a blanket cachebust here, to get restore _obviously_ working.

Reply:
this seems fine as a temporary solution. One potentially simpler alternative would be to not set a cache duration of zero here in the case where no version id is specified: https://github.com/code-dot-org/code-dot-org/blob/e76bcc76e1d6fb7cfbdc450b4bcf8e2593f688b7/shared/middleware/files_api.rb , although perhaps your solution is more specific in that it allows caching of unversioned animation urls in other situations I'm not thinking of?

Reply:
given that you've already written tests for this, I'll re-emphasize that the way you have this seems fine for now :-)

Reply:
:+1: I'm going to stick with this, especially since I think we'll tear it back out soon.  I think one upside of this approach is that it'll work immediately after we deploy this change for all clients, including those that have existing animations locally cached before we introduced the `no_cache` directive.